src/checkers/player.js

```diff
@@ -0,0 +1,46 @@
+import { Action } from './action.js';
+export class Player {
+    // Private fields
+    #name;
+
+    #pieces;
+
+    #pieceColor;
+
+    #pieceSelected;
+
+    constructor(name, pieceColor, pieces) {
+        this.#name = name;
+        this.#pieceColor = pieceColor;
+        this.#pieces = pieces;
+        this.#pieceSelected = null;
+    }
+
+    get name() {
+        return this.#name;
+    }
+
+    get pieceColor() {
+        return this.#pieceColor;
+    }
+
+    get pieceSelected() {
+        return this.#pieceSelected;
+    }
+
+    set pieceSelected(piece) {
+        this.#pieceSelected = piece;
+    }
+
+    getPiece(piecePosition) {
+        return this.#pieces.get(piecePosition);
+    }
+
+    selectPiece(piecePosition) {
+        Action.selectPiece(this, piecePosition);
+    }
```

Comment:
The way I see it, action.js should be a file that contains just functions not a Class, because I think, if it has no attributes(privates fields) it's not necessary create a class, at least for me, I wait suggestions.